### PR TITLE
discordo: init at unstable-2023-04-07

### PR DESCRIPTION
**Pull request by Arian-D**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by lilyinstarlight — APPROVED** [review]
This seems to build just fine for me on x86_64-linux (though I did not actually try to run it) and the derivation looks good

**Comment by Arian-D** [comment]
Just a mini update on this:
- I tested it on aarch64 (non-nixos)
- Updated the commit to a more recent version.

**Comment by spikespaz** [comment]
What about [clipboard support](https://github.com/ayn2op/discordo#linux-clipboard-support)?

Should add them to `buildInputs`.

**Comment by lilyinstarlight** [comment]
> What about [clipboard support](https://github.com/ayn2op/discordo#linux-clipboard-support)?
> 
> Should add them to `buildInputs`.

Adding them to `buildInputs` won't do anything because they are needed at runtime. We could `wrapProgram` the main binary in the derivation to always have those in `PATH`, though

Something like:

```nix
buildGoModue rec {
  #stuff...
  nativeBuildInputs = [ makeWrapper ];
  #stuff...
  postInstall = ''
    wrapProgram $out/bin/discordo \
      --prefix PATH : ${lib.makeBinPath [ xsel wl-clipboard ]}
  '';
}
```

**Review by adamcstephens — APPROVED** [review]
Looks good on darwin, too. This commit is now a month behind upstream.

Result of `nixpkgs-review pr 210667` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>discordo</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 210667` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>discordo</li>
  </ul>
</details>

**Comment by lilyinstarlight** [comment]
Do you think you could bump this to a newer commit? I think it's probably merge-ready

Also if you want to learn something cool for automating the updates, you can add `unstableGitUpdater` to the file args and add `passthru.updateScript = unstableGitUpdater { };` to the derivation, and then r-ryantm should occasionally pick up updates automatically and you can run the update script manually like `nix-shell maintainers/scripts/update.nix --argstr commit true --argstr package discordo`

File: pkgs/applications/networking/discordo/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ayn2op";
+    repo = pname;
+    rev = "b6524689788345cf265d584cd37cee734d220d61";
+    hash = "sha256-Y3O41r6pxRcePOhszgDjXI4tCtArgD9sTkxTZwIWtMA=";
+  };
+
+  vendorHash = "sha256-2xTGdwfQlHgprrJ1xEBSnlDHTwNMjGnhEKN1hgVg5So=";
+
```
Comment (Kranzes) [inline pkgs/applications/networking/discordo/default.nix:15]:
```suggestion

  ldflags = [ "-w" "-s" ];
  
```

**Comment by Arian-D** [comment]
I apologize about the delay; for some reason I was not getting notification until the most recent message.
- Per spikespaz's requeset I added `xsel` and `wl-clipboard` to the definition
- I added the `ldflags` per Kranzes' suggestion
- I bumped it up to a more recent version and added the unstable updater. Thank you lilyinstarlight for the guidance and advice

**Comment by adamcstephens** [comment]
You'll want to update the title of the PR to match the new commit description.

This also builds and runs successfully as a static binary, so I'd suggest adding `CGO_ENABLED = 0;`

**Comment by lilyinstarlight** [comment]
> * I bumped it up to a more recent version and added the unstable updater. Thank you lilyinstarlight for the guidance and advice

So I think I didn't notice it was a Go module when I said that, you actually need to use nix-update (instead of `unstableGitUpdater`) so that the vendorHash gets updated too:

```
  passthru.updateScript = nix-update-script {
    extraArgs = [ "--version=branch" ];
  };
```

File: pkgs/applications/networking/discordo/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  src = fetchFromGitHub {
+    owner = "ayn2op";
+    repo = pname;
+    rev = "f8c58057945b1ded2f78dc0956ea25aa281a0b31";
+    hash = "sha256-FUSPQK4rB0J89s+I7yhF8SQ/Q9uygQSCF9o6ltYxOk4=";
+  };
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [ "--version=branch" ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/discordo/default.nix:17]: passthru is usually put right before meta

File: pkgs/applications/networking/discordo/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/discordo \
+      --prefix PATH : ${lib.makeBinPath [ xsel wl-clipboard ]}
+  '';
+
+  meta = with lib; {
+    description =
+      "A lightweight, secure, and feature-rich Discord terminal client";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/discordo/default.nix:35]:
```suggestion
    description = "Lightweight, secure, and feature-rich Discord terminal client";
```

Comment (Arian-D) [reply pkgs/applications/networking/discordo/default.nix]: What formatter would you suggest? Emacs (using `nixfmt` I believe) formats long lines like that

Comment (SuperSandro2000) [reply pkgs/applications/networking/discordo/default.nix]: nixpkgs-fmt or change the line length of nixfmt to 120 or something similar to not generate such silly short lines.

**Pull request merged**